### PR TITLE
PyQt5 port

### PR DESCRIPTION
**Pull request by rkern**
I have ported this to PyQt5. In so doing, I also dropped Python 2.7 (and PySide as it is 2-only) and 3.5 from the test configurations. We're 3.6+ here on out.

As noted in the commit message for ef19914, there's a minor behavior change due to PyQt5 dropping some of the meta-programming tools I used for signals. But it's minor and ultimately desirable.

### CODE REVIEW

**Comment by rkern** [comment]
Fixes #57

File: qt_binder/tests/test_binder.py [hunk 1]
```diff
@@ -193,19 +190,22 @@ def handler(new):
         # Check delayed signal.
         qobj.destroyed[QtCore.QObject].emit(qobj)
 
-        six.assertCountEqual(self, received, [(), qobj])
+        # When there are 2 overloads for a signal and one of the is
+        # argumentless, we actually get the single-argument version regardless.
+        # This is actually typically what we want, but I'm noting it because
+        # this behavior did change when we redid the signal implementation to
+        # conform to the limitations of the PyQt5 signal implementation.
+        six.assertCountEqual(self, received, [qobj])
```
Comment (jvkersch) [inline qt_binder/tests/test_binder.py:198]: Do you still need `six` here, given that the library is now Python-3 only?

**Review by jvkersch — APPROVED** [review]
Code-wise LGTM, just a question about `six`.

I didn't get to try this out on a realistic app, though. I will try to get that done in the next few days.

**Comment by rahulporuri** [comment]
I tested the changes in this branch with PyQt5 version 5.7.1 on the canopy geoscience application. On master, the application and tests fail and the app/tests work as expected on this branch.

I am seeing interesting messages on the console related to `QObjects`. I'm not sure if they're related to qt_binder or they're because of what canopy geo is doing. See example below

```
$ canopy-geoscience
QWidgetWindow(0x101dc5470, name="QStackedWidgetClassWindow") ( QScreen(0x12f23aed0, name="Color LCD") ): Attempt to set a screen on a child window.
QWidgetWindow(0x101dc5470, name="QStackedWidgetClassWindow") ( QScreen(0x12f23aed0, name="Color LCD") ): Attempt to set a screen on a child window.
QWidgetWindow(0x101dc5410, name="EditorAreaWidgetClassWindow") ( QScreen(0x12f23aed0, name="Color LCD") ): Attempt to set a screen on a child window.
```

```

File: qt_binder/tests/test_binder.py (hunk 1 above)
Comment (rkern) [reply qt_binder/tests/test_binder.py:198]: No, but I prefer to do that in a cleanup PR separate from the new feature, for reviewer's sanity.

**Comment by jvkersch** [comment]
Somewhat belatedly, but testing on our app in the wild revealed no problems (with QT binder at least).

**Pull request merged**